### ISSUE TITLE
Updates to CloudFront distribution wait indefinitely for completion

### ISSUE THREAD

**Issue by lukemundy-vgw** [pulumi/pulumi-aws#857]
I'm having an issue with CloudFront distributions, @clstokes  [suggested I open an issue for it](https://pulumi-community.slack.com/archives/CRH5ENVDX/p1579705440019900?thread_ts=1579678176.018600&cid=CRH5ENVDX).

I've been working on building a CloudFront distribution and having an issue where both the creation and update of the distribution waits indefinitely for completion. For example, I would make a change (any change), run `pulumi up`, the change would be made and the distribution moves into the `In Progress` state. During this time, as expected, Pulumi shows the status as `updating`. Once the change is complete though and the distribution has returned to the `Deployed` state, Pulumi continues waiting for the update to complete.

This occurred every time I made an update to the distribution (approximately 4-5 times). I left it running long enough that the AWS credentials in use expired (1 hour) at which point Pulumi terminates with a credential error.

I tried working around the issue by providing a `customTimeout` on updates of 15m, however this appeared to have no effect.

Versions:
- AWS Typescript Project
- MacOS Catalina
- Node v10.12.0
- @pulumi/aws v1.18.0
- @pulumi/pulumi v1.8.1
- Pulumi CLI v1.8.1

Relevant part of the template:
```typescript
const cdn = new aws.cloudfront.Distribution("cdn", {
    comment: serviceName,
    enabled: true,
    aliases: [ `${cfg.require("cdnName")}.${rootDomainName}` ],
    viewerCertificate: {
        sslSupportMethod: "sni-only",
        acmCertificateArn: cfg.require("certificateArn"),
        minimumProtocolVersion: "TLSv1.1_2016"
    },
    priceClass: "PriceClass_100",
    defaultRootObject: "index.html",
    isIpv6Enabled: true,
    customErrorResponses: [
        { errorCode: 400, errorCachingMinTtl: 60, responseCode: 400, responsePagePath: "/errors/400.html" },
        { errorCode: 403, errorCachingMinTtl: 60, responseCode: 403, responsePagePath: "/errors/403.html" },
        { errorCode: 404, errorCachingMinTtl: 60, responseCode: 404, responsePagePath: "/errors/404.html" },
        { errorCode: 500, errorCachingMinTtl: 60, responseCode: 500, responsePagePath: "/errors/500.html" },
        { errorCode: 502, errorCachingMinTtl: 60, responseCode: 502, responsePagePath: "/errors/502.html" },
        { errorCode: 503, errorCachingMinTtl: 60, responseCode: 503, responsePagePath: "/errors/503.html" },
        { errorCode: 504, errorCachingMinTtl: 60, responseCode: 504, responsePagePath: "/errors/504.html" },
    ],
    origins: [
        {
            originId: "default",
            domainName: bucket.bucketRegionalDomainName,
            s3OriginConfig: {
                originAccessIdentity: oai.cloudfrontAccessIdentityPath
            }
        }
    ],
    defaultCacheBehavior: {
        targetOriginId: "default",
        allowedMethods: [
            "DELETE",
            "GET",
            "HEAD",
            "OPTIONS",
            "PATCH",
            "POST",
            "PUT",
        ],
        cachedMethods: [
            "GET",
            "HEAD",
        ],
        viewerProtocolPolicy: "redirect-to-https",
        defaultTtl: 3600,
        compress: true,
        forwardedValues: {
            cookies: { forward: "none" },
            queryString: false
        },
    }
}, {
    customTimeouts: {
        update: "15m"
    }
});
```

Let me know if you'd like any further information.

**Comment by lukemundy-vgw** [comment]
I've tried another small update this morning to troubleshoot. I can see in my CloudTrail logs that Pulumi sends the `UpdateDistribution` request at 2020-01-23T03:24:16Z and then several (I counted 70) `GetDistribution` requests every ~10 seconds or so until 2020-01-23T03:38:22Z when they stop. There are no further requests to AWS after that point, however Pulumi is still waiting for the update to complete.

I can't be sure exactly what time the distribution returned to the `Deployed` state, but its possible it was after 2020-01-23T03:38:22Z which may be why Pulumi never noticed. Perhaps the underlying terraform provider stops sending `GetDistribution` requests after ~15 minutes?

**Comment by lukemundy-vgw** [comment]
Ok after some further troubleshooting I found what the issue likely is. I tracked down the `AssumeRole` event that generated the credentials for Pulumi to use and saw that the session duration was set to 900 seconds (not an hour as I first thought).

So it appears in this case, the credentials expired but Pulumi (or the underlying terraform provider) seems to not notice and hangs. The command has been running for over an hour now with no error.

I'll try again with a longer session duration but it'd be nice if expired credentials caused errors immediately.

**Comment by lukemundy-vgw** [comment]
Ok longer duration credentials worked fine so the only issues still remaining are the `customTimeout` not working and the terraform provider not noticing that credentials have expired.

**Comment by pgavlin** [comment]
@stack72 any ideas?

**Comment by lukehoban** [comment]
> I tracked down the AssumeRole event that generated the credentials for Pulumi to use and saw that the session duration was set to 900 seconds (not an hour as I first thought).

I believe this is the root issue, and is a duplicate of https://github.com/pulumi/pulumi-aws/issues/2205.  Will close this to track there.
